src/__tests__/within.test.js

```diff
@@ -0,0 +1,72 @@
+// @flow
+import React from 'react';
+import { View, Text, TextInput } from 'react-native';
+import { render } from '..';
+import { within } from '../within';
+
+test('within() exposes basic queries', () => {
+  const rootQueries = render(
+    <View>
+      <View testID="first">
+        <Text>Same Text</Text>
+        <TextInput value="Same Value" placeholder="Same Placeholder" />
+      </View>
+      <View testID="second">
+        <Text>Same Text</Text>
+        <TextInput value="Same Value" placeholder="Same Placeholder" />
+      </View>
+    </View>
+  );
+
+  expect(rootQueries.getAllByText('Same Text')).toHaveLength(2);
+  expect(rootQueries.getAllByDisplayValue('Same Value')).toHaveLength(2);
+  expect(rootQueries.getAllByPlaceholder('Same Placeholder')).toHaveLength(2);
+
+  const firstQueries = within(rootQueries.getAllByTestId('first')[0]);
```

Comment:
Can we avoid using ByTestId in our tests?

Reply:
Replaced byTestId with byA11yHint